waves/src/App.test.tsx

```diff
@@ -150,25 +150,34 @@ test("Swap asset types", () => {
         beta: AssetType.USDT,
     };
 
-    let newValue = AssetType.BTC;
     expect(
         reducer(initialState, {
             type: "SwapAssetTypes",
+            value: {
+                betaAmount: initialState.alpha.amount,
```

Comment:
Not sure if this test makes a whole lot of sense if you pass in the original value of `alpha.amount`?

The reducer no longer calls `calculateBetaAmount`. Hence, the current behaviour can be tested with a much simpler and easier to understand test (simply assert that the value passed in the action gets set to `alpha.amount`).

Reply:
For me this makes a whole lot of sense as it shows that alpha and beta amounts are swapped afterwards. :) 

Let me know what you don not understand here please and we can change the test. :)

Reply:
> For me this makes a whole lot of sense as it shows that alpha and beta amounts are swapped afterwards. :)

How much value does the assertion have if there is not observable difference because the numbers are the same? :D

The reducer sets `alpha.amount` to `betaAmount`. If you pass `state.alpha.amount` as `betaAmount` there is no effective change of the value of `alpha.amount`.

Hence the assertion is weak because removing the assignment from within the reducer would not actually break the test.

`SwapAssetTypes` does two concrete things:

- swap the types of the amounts
- assign `betaAmount` to `state.alpha.amount`

Given that, a clear test should:

1. Initialize the state (ideally in a concise way) that shows which elements of the state we care about: `alpha.type`, `alpha.amount` & `beta.type`
2. Dispatch `SwapAssetTypes` with a hard-coded number where it is clearly visible that it is different from `alpha.amount`
3. Assert that the asset types are flipped and that `betaAmount` is the new `alpha.amount`

Reply:
> Given that, a clear test should:

> 
>     Initialize the state (ideally in a concise way) that shows which elements of the state we care about: alpha.type, alpha.amount & beta.type
>     Dispatch SwapAssetTypes with a hard-coded number where it is clearly visible that it is different from alpha.amount
>     Assert that the asset types are flipped and that betaAmount is the new alpha.amount


Imho that's precisely what I'm doing. :) 

1) First few lines of the test setup the sate: 
https://github.com/comit-network/droplet/blob/d7c99d65dd47e8f1fda67df9422aa090d5f2bee7/waves/src/App.test.tsx#L136-L143

2) Dispatched the `SwapAssetType` with a fixed amount (I slightly changed it in the last patch)

https://github.com/comit-network/droplet/blob/d7c99d65dd47e8f1fda67df9422aa090d5f2bee7/waves/src/App.test.tsx#L151

3) Assert that the types are flipped and the `betaAmount` as specified in the event is assigned to `alpha.amount`.